packages/cadl-ranch-specs/http/models/property-types/mockapi.ts

```diff
@@ -23,6 +23,12 @@ function createMockApis(route: string, value: any): MockApiGetPut {
       };
     }),
     put: mockapi.put(url, (req) => {
+      if (route === "datetime" && new Date(body.property).toISOString() === new Date(req.body.property).toISOString()) {
+        return {
+          status: 204,
+        };
+      }
+
       req.expect.bodyEquals(body);
```

Comment:
If we do the comparison in using ISOString, do we still need this `req.expect.bodyEquals(body)`?

Reply:
I think in that case it would already have returned before(line 28). Think for a better design we could have an additional argument to `createMockApis` called `convertBody` and it would call that when comparing the body instead